lib/core/MantraData.js

```diff
@@ -0,0 +1,303 @@
+    return !isNaN && !isNull;
+};
+
+/**
+ * Class representing a common configuration for MantraComponents.
+ */
+export default class MantraData {
+    /**
+     * Creates a MantraData object with default value in properties.
+     */
+    constructor({ path, baseClass }) {
+        this.path = '';
+        this.component = null;
+        this.parents = [];
+        this.name = '';
+        this.schema = null;
+        this.id = '';
+        this.action = '';
+        this.context = '';
+        this.endpoint = '';
+        this._baseClass = baseClass;
+        this._path = path;
+    };
+
+    /**
+     * Sets the name and schema property to the value of the entity in the _path property.
+     * Updates the path property and trimms the _path property.
+     */
+    setBase() {
+        const name = MantraData.getFirstPath(this._path);
+        if (!this._isSchema(name)) {
+            throw new Error(`The ${name} schema was not found`);
+        }
+
+        this._setName(name);
+        this._setBaseSchema(name);
+        this.updatePath(name);
+    };
+
+    /**
+     * Sets the class name property to the entity's name.
+     * @param {string} name - The entity's name.
+     */
+    _setName(name) {
+        this.name = name;
+    };
+
+    /**
+     * Verify the existence of the name paramter within the schema in the store.
+     * @param {string} name - The name of a possible schema.
+     * @returns {boolean} - True if the schema exists in the store.
+     */
+    _isSchema(name) {
+        const schemas = MantraPlugin.getSchemasModule();
+
+        if (schemas.hasOwnProperty(name)) 
+            return true;
+        else 
+            return false;
+    };
+
+    /**
+     * Sets the class schema property to the entity's schema object.
+     * @param {string} schema - The entity's schema name.
+     */
+    _setBaseSchema(schema) {
+        const schemas = MantraPlugin.getSchemasModule();
+        this.schema = schemas[schema];
+    };
+
+    /**
+     * Set the id property and may re-set the schema, name and id property.
+     */
+    setRecord() {
+        const arrPath = this._path.split('.');
+
+        for (const name of arrPath) {
+            if (this._isRelationship(name)) {
+                this._setRelationship(name);
+                continue;
+            }
+            
+            if (this._isId(name)) {
+                this._setId(name);
+                continue;
+            }
+
+            break;
+        }
+    };
+
+    /**
+     * Verify the existence of the relationship name inside the schema property.
+     * @param {string} name - The relationship name to be validated.
+     * @returns {boolean} - True if the relationship name exists in the schema property.
+     */
+    _isRelationship(name) {
+        const relationships = this.schema.relationships;
+        const HAS_RELATIONSHIPS = (relationships !== undefined);
+        return HAS_RELATIONSHIPS && relationships.some((rel) => rel.name === name);;
```

Comment:
I tried to create a constant for the "some" evaluation but I ineffectively couldn't achieve it. 

The problem is that the "some" method expected the "relationships" to be always an array and if not it will throw an exception.

But I could create a constant anyways but that would imply that if the "relationships" constant is not an array then an exception will be thrown. (Just for us to be aware of that).